sim.c

```diff
@@ -765,19 +765,21 @@ BEGIN_OPERATOR(zig)
   LOWERCASE_REQUIRES_BANG;
   Glyph* gline = gbuffer + width * y;
   gline[x] = '.';
-  if (x + 1 == width)
-    return;
-  if (gline[x + 1] == '.') {
+  if (gline[x + 1] == '.' && x + 1 != width) {
     gline[x + 1] = This_oper_char;
     mbuffer[width * y + x + 1] |= (U8)Mark_flag_sleep;
   } else {
-    Usz n = 256;
+    Usz n = 512;
     if (x < n)
       n = x;
     for (Usz i = 0; i < n; ++i) {
-      if (gline[x - i - 1] != '.') {
+      Usz search_idx = x - i - 1;
+      if (gline[search_idx] != '.') {
         gline[x - i] = This_oper_char;
         break;
+      } else if (search_idx == 0) {
```

Comment:
This loop is not well-formed. This adds an unnecessary extra branch on each iteration. You will need to rewrite this to some other form.